pool/tests/nrd_kernel_relative_height.rs

```diff
@@ -99,7 +105,7 @@ fn test_nrd_kernel_relative_height() -> Result<(), PoolError> {
 			&keychain,
 			vec![24],
 			vec![18],
-			kernel.clone(),
+			kernel2.clone(),
```

Comment:
Actually this (and the creation of the 2nd kernel above) is necessary because of the changes to tx validation.
This was actually a poorly constructed test as we were reusing a kernel multiple times.
We used to get away with this but technically we cannot have duplicate kernels in a tx and changes in this PR surfaced this issue.